framework/doc/sqa/SoftwareRequirements.tex

```diff
@@ -0,0 +1,454 @@
+\documentclass{INLreport}
+\SANDnum{MOOSE-SRS}
+\SANDprintDate{February 2016}
+\SANDreleaseType{Unlimited Release}
+\SANDauthor{Cody Permann}
+\title{Multiphysics Object Oriented Simulation Environment (MOOSE) Software Requirements Specification and Software Design Description}
+
+\usepackage[table]{xcolor}
+
+\usepackage{caption}
+\captionsetup{labelsep=space,justification=justified,labelfont=bf,singlelinecheck=off}
+
+% Use for numbering rows of tables.
+\newcommand{\TableRowNum}[1]{\thetable.\arabic{#1}\stepcounter{#1}}
+
+\begin{document}
+\maketitle
+
+\SANDmain
+
+\section{Introduction}
+\subsection{Purpose}
+The Multiphysics Object-Oriented Simulation Environment (MOOSE) is a tool for solving complex coupled Multiphysics equations
+using the finite element method. MOOSE uses an object-oriented design to abstract data structure managment, parallelism,
+threading and compiling while providing an easy to use interface targeted at engineers that may not have a lot of
+software development experience. MOOSE will require extreme scalability and flexibility when compared to other FEM
+frameworks. For instance, MOOSE needs the ability to run extremely complex material models, or even third-party applications
+within a parallel simulation without sacrificing parallelism. This capability is in contrast to what is often seen in
+commercial packages, where custom material models can limit the parallel scalability, forcing serial runs in the most severe
+cases. When comparing high-end capabilities, many MOOSE competitors target modest-sized clusters with just a few thousand
+processing cores. MOOSE, however, will be required to routinely executed on much larger clusters with scalability to clusters
+available in the top 500 systems. (top500.org). MOOSE will also be targeted at smaller systems such as high-end laptop
+computers.
+
```

Comment:
Agreed, This is part of a general overview of the the project. I don't have any requirements tied to scalability at this point. We may add that in the future if we want to architect a way to test this requirement.